### ISSUE TITLE
Bug: server url changes are not saving

### ISSUE THREAD

**Issue by yantato** [mattermost/focalboard#4439]
## Steps to reproduce the behavior

We changed our mattermost server's url, and whenever I try to change it, after attempting to log in, it reverts to the old url

## Expected behavior

Server url should not be reverting to the previous url when saving server config

## Edition and Platform

 - Edition: Personal Desktop
 - Version: 5.2.2
 - Browser and OS: Chrome on Windows 10

## Additional context (optional)

- I've tried reinstalling, same issue
- Another machine's desktop app works fine, same machine's browser app works fine; other colleagues are not experiencing the same issue on their machines
- I'm using tailscale vpn (not sure if this matters)

**Comment by wuwinson** [comment]
@yantato Are you referring to Mattermost Boards or Channels or something else? Can you also provide some screenshots for reference? Thanks! cc @sbishel

**Comment by yantato** [comment]
> @yantato Are you referring to Mattermost Boards or Channels or something else? Can you also provide some screenshots for reference? Thanks! cc @sbishel

mattermost channels

1. I edit the server info here
![image](https://user-images.githubusercontent.com/62998387/211440187-84c6d1a7-58e4-47a9-a229-a7297229bb70.png)

2. This is the current url
![image](https://user-images.githubusercontent.com/62998387/211440355-61767e36-f8b7-4f27-a3aa-e278967ccfb5.png)

3. Change it to the new url, save it, mattermost reloads
![image](https://user-images.githubusercontent.com/62998387/211440409-36523368-4f51-4bf9-bb36-fb96e8293202.png)

4. But still gives an error message
![image](https://user-images.githubusercontent.com/62998387/211440478-90535c7d-df84-4422-bbc8-592f992a969e.png)

5. Then i check the server info again and the url remains unchanged even after saving on step 3
![image](https://user-images.githubusercontent.com/62998387/211440558-5c72b0ca-cb48-4771-816f-3e9e2ab0dac3.png)

**Comment by wuwinson** [comment]
Thanks for clarifying @yantato! @esethna @amyblais This looks like a Channels bug, do you want me to transfer this over to the MM repro?

**Comment by amyblais** [comment]
Yes, please move this to the desktop repo https://github.com/mattermost/desktop/issues. cc @yantato
